core/src/test/java/org/apache/sdap/mudrod/weblog/structure/log/GeoIpTest.java

```diff
@@ -0,0 +1,18 @@
+package org.apache.sdap.mudrod.weblog.structure.log;
+
+import static org.junit.Assert.*;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class GeoIpTest {
+
+	@Test
+	public void testToLocation() {
+		GeoIp ip = new GeoIp();
+		String iptest = "185.10.104.194";
+		Coordinates result = ip.toLocation(iptest);
+		Assert.assertEquals("failed in geoip function!", "22.283001,114.150002", result.latlon);
+	}
+
+}
```

Comment:
We cannot ship binary code inside of source code management... the resources below should be decompressed.

Reply:
Also, is there any reason we are using logs from 2015? Why not 2018?

Reply:
I cannot understand which resource do you mean should be decompressed?  There is not a specific reason for choosing logs from 2015.

Reply:
I remembered that @fgreg has added a function of decompressing zip log file for MUDROD. And the test code can parse the two gz. file. If you need, I can depress them for you.  Is it necessary?